### ISSUE TITLE
Add option to allow specified actors to bypass required pull requests

### ISSUE THREAD

**Issue by jasminjenull** [repository-settings/app#676]
## Prerequisites:

The feature is available in the Github UI under `settings/branch_protection_rules`.
<img width="522" alt="image" src="https://user-images.githubusercontent.com/117167929/207917372-9058c803-42ab-4b38-bc80-8ba54991cb03.png">

It is also available through the Github API and can be found [here](https://docs.github.com/en/rest/branches/branch-protection?apiVersion=2022-11-28) under `bypass_pull_request_allowances`.

## New Feature

We are currently having this in our `settings.yml`:

```
branches:
- name: master
  protection:
    required_pull_request_reviews:
      required_approving_review_count: 1
      dismiss_stale_reviews: true
      require_code_owner_reviews: true
      dismissal_restrictions:
        users:
          - semantic-release-bot
        teams: []
    required_status_checks: null
    enforce_admins: true
    required_linear_history: true
    restrictions:
      apps: []
      users:
        - semantic-release-bot
      teams: []
```

With these settings semantic-release is not able to create releases on the master branch:

```
stderr: 'remote: error: GH006: Protected branch update failed for refs/heads/master.        \n' +
    'remote: error: Waiting on code owner review from xxx/xxx.        \n' +
    'To [https://github.com/xxx/xxx\n](https://github.com/xxx/xxx/n)' +
    ' ! [remote rejected] HEAD -> master (protected branch hook declined)\n' +
    "error: failed to push some refs to 'https://github.com/xxx/xxx'",
```

When enabling that checkbox to allow specified actors to bypass required pull requests manually in the UI, it succeeds. Would be nice to have that functionality available within our config to avoid manual changes.

Or perhaps there's also another way to solve this? 🤔

**Comment by travi** [comment]
if the property is available through the api, it should work through the settings app. have you tried adding the property to your settings config?

> Or perhaps there's also another way to solve this? 🤔

also, as a maintainer of semantic-release, i would suggest that it is worth considering if you need the release to produce a commit that needs to be pushed in the first place. if you use github releases (or similar) for your release notes rather than a changelog, and consider the version being stored as a tag rather than also in your package manifest being good enough, then you could potentially avoid the situation altogether. this does assume you arent making commits for other reasons outside the common usecase, but just highlighting that it is worth consideration.

**Comment by jasminjenull** [comment]
thanks for your quick response! 🥇 I've tried adding it like this:
```
branches:
- name: master
  protection:
    required_pull_request_reviews:
      required_approving_review_count: 1
      dismiss_stale_reviews: true
      require_code_owner_reviews: true
      dismissal_restrictions:
        users:
          - semantic-release-bot
        teams: []
      bypass_pull_request_allowances:
        users:
          - semantic-release-bot
        teams: []
```
Also gave it a try to remove one indentation and placing it directly under `protection` but it doesn't seem to have any effect. Should it be done in a different way? 🤔

Ah yeah, true. I think having it added to a changelog file is still preferred for us but maybe we can think of using release notes instead 👍

**Comment by paddyroddy** [comment]
Should this be
```yaml
bypass_pull_request_allowances:
  apps:
    - semantic-release-bot
```
Just trying this myself @j-jasmin
